[CI] GoogleCloudStorageBlobStoreRepositoryTests testRequestStats failing

**Build Scans:**
- [elasticsearch-intake #21905 / part1](https://gradle-enterprise.elastic.co/s/rpfbumybmujgi)
- [elasticsearch-pull-request #68137 / part-1](https://gradle-enterprise.elastic.co/s/xp2fpc2wb3riq)
- [elasticsearch-pull-request #68143 / part-1](https://gradle-enterprise.elastic.co/s/qatlgln5xuqo2)
- [elasticsearch-pull-request #68138 / part-1](https://gradle-enterprise.elastic.co/s/ojvocpgtq4msc)
- [elasticsearch-pull-request #68137 / part-1](https://gradle-enterprise.elastic.co/s/3dk7fgpbdcu7g)
- [elasticsearch-pull-request #68132 / part-1](https://gradle-enterprise.elastic.co/s/zcj2jcn7s56ze)
- [elasticsearch-pull-request #68130 / part-1-fips](https://gradle-enterprise.elastic.co/s/be64tpqynedfu)
- [elasticsearch-pull-request #68130 / part-1](https://gradle-enterprise.elastic.co/s/p3qn6djrigzas)
- [elasticsearch-pull-request #68125 / part-1](https://gradle-enterprise.elastic.co/s/zfp5rxfkrijei)

**Reproduction Line:**
```
./gradlew ":modules:repository-gcs:internalClusterTest" --tests "org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStoreRepositoryTests.testRequestStats" -Dtests.seed=93BC0B8AE3162E96 -Dtests.locale=fr-BF -Dtests.timezone=Australia/Canberra -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStoreRepositoryTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testRequestStats),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a value greater than <0>
     but: <0> was equal to <0>
```

**Issue Reasons:**
- [main] 9 failures in test testRequestStats (4.9% fail rate in 182 executions)
- [main] 7 failures in step part-1 (8.8% fail rate in 80 executions)
- [main] 6 failures in pipeline elasticsearch-pull-request (7.4% fail rate in 81 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/repository-gcs/src/internalClusterTest/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java
+++ b/modules/repository-gcs/src/internalClusterTest/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java
@@ -343,7 +343,15 @@ public class GoogleCloudStorageBlobStoreRepositoryTests extends ESMockAPIBasedRe
             }
 
             if (exchange.getRequestHeaders().containsKey(IDEMPOTENCY_TOKEN)) {
-                return exchange.getRequestHeaders().getFirst(IDEMPOTENCY_TOKEN);
+                String idempotencyToken = exchange.getRequestHeaders().getFirst(IDEMPOTENCY_TOKEN);
+                // In the event of a resumable retry, the GCS client uses the same idempotency token for
+                // the retry status check and the subsequent retries.
+                // Including the range header allows us to disambiguate between the requests
+                // see https://github.com/googleapis/java-storage/issues/3040
+                if (exchange.getRequestHeaders().containsKey("Content-Range")) {
+                    idempotencyToken += " " + exchange.getRequestHeaders().getFirst("Content-Range");
+                }
+                return idempotencyToken;
             }
 
             final String range = exchange.getRequestHeaders().getFirst("Content-Range");

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -405,18 +405,6 @@ tests:
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=ml/inference_crud/Test delete given unused trained model}
   issue: https://github.com/elastic/elasticsearch/issues/126881
-- class: org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStoreRepositoryTests
-  method: testRequestStats
-  issue: https://github.com/elastic/elasticsearch/issues/126902
-- class: org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStoreRepositoryTests
-  method: testWriteReadLarge
-  issue: https://github.com/elastic/elasticsearch/issues/126903
-- class: org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStoreRepositoryTests
-  method: testWriteFileMultipleOfChunkSize
-  issue: https://github.com/elastic/elasticsearch/issues/126908
-- class: org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStoreRepositoryTests
-  method: testSnapshotWithLargeSegmentFiles
-  issue: https://github.com/elastic/elasticsearch/issues/126909
 - class: org.elasticsearch.index.engine.CompletionStatsCacheTests
   method: testCompletionStatsCache
   issue: https://github.com/elastic/elasticsearch/issues/126910

